### ISSUE TITLE
Rundeck refusing to kill job after dead process

### ISSUE THREAD

**Issue by miversen33** [rundeck/rundeck#9233]
**Describe the bug**
When a job fails in rundeck, it stays stuck in a running state and becomes "unkillable". 

![image](https://github.com/user-attachments/assets/e72b974e-58e3-4126-a3a1-c64c38b9dc58)
Below is the job configuration here. I don't much care about the error I am seeing, its likely a misconfiguration of the job itself. However, I _cannot_ kill the job to begin troubleshooting it

[Clear_OIDP_Instance_Data_for_Service_Users.json](https://github.com/user-attachments/files/16250541/Clear_OIDP_Instance_Data_for_Service_Users.json)

**My Rundeck detail**

* Rundeck version: 5.4.0
* install type: docker
* OS Name/version: os the rundeck/rundeck:SNAPSHOT image runs on
* DB Type/version: postgresql

I am using a slightly modified version of the docker compose configuration found in [docker-zoo](https://github.com/rundeck/docker-zoo/blob/master/postgres/docker-compose.yml)

```yaml
services:
  db:
    image: postgres:latest
    environment:
      - POSTGRES_PASSWORD=${PGPASS?Missing PGPASS}
      - POSTGRES_USER=${PGUSER?Missing PGUSER}
    volumes:
      - db:/var/lib/postgresql/data
      - /opt/docker/rundeck/dbinit.sql:/docker-entrypoint-initdb.d/01-dbinit.sql
    restart: "unless-stopped"
  rundeck:
    image: rundeck/rundeck:SNAPSHOT
    ports:
      - 4430:4440
    depends_on:
      - db
    environment :
      - RUNDECK_SERVER_CONTEXTPATH=/cron
      - RUNDECK_MAIL_SMTP_HOST=hostname
      - RUNDECK_MAIL_SMTP_PORT=1025
      - RUNDECK_MAIL_FROM=cron@myhost
      - RUNDECK_GRAILS_URL=${RUNDECK_GRAILS_URL:-http://localhost:4440}
      - RUNDECK_SECURITY_SYNCLDAPUSER=true
      - RUNDECK_DATABASE_DRIVER=org.postgresql.Driver
      - RUNDECK_DATABASE_USERNAME=${RD_USER:-rundeck}
      - RUNDECK_DATABASE_PASSWORD=${RD_PASSWORD:-password}
      - RUNDECK_DATABASE_URL=jdbc:postgresql://db/rundeck
    volumes:
      - rundeck_data:/home/rundeck/server/data
      - rundeck_server:/home/rundeck/server/
      - /opt/ipg/backups:/mnt/backups
      - /opt/docker/rundeck/jaas-loginmodule.conf:/home/rundeck/server/config/jaas-loginmodule.conf
      - /opt/docker/rundeck/cacerts:/etc/ssl/certs/java/cacerts
      - /opt/docker/rundeck/nodes:/home/rundeck/server/extra_nodes
    restart: "unless-stopped"

    extra_hosts:
      - "host.docker.internal:host-gateway"

volumes:
  rundeck_data:
  rundeck_server:
  db:
```

The contents of `dbinit.sql` are
```sql
CREATE USER rundeck WITH PASSWORD 'password';
CREATE DATABASE rundeck WITH OWNER rundeck;
GRANT ALL PRIVILEGES ON DATABASE rundeck TO rundeck;
```

**To Reproduce**
- Import the attached job (or create any job that will fail before it starts)
- Run said job
- Attempt to either "Mark as Incomplete" or "Kill Job"

**Expected behavior**
I would expect 
1) Rundeck would notice that the job it spawned is dead and act accordingly
2) That I can forcibly tell rundeck the job is dead, stop tracking it

**Screenshots**

**Desktop (please complete the following information):**
 - OS: Windows 10
 - Browser: Firefox/Chrome

**Additional context**
I noticed that when I click "Mark as Incomplete" or "Kill Job", the endpoint _never_ responds. It just sits there. I let the request sit for 5 minutes and never see any response. 

![image](https://github.com/user-attachments/assets/ba5727ad-dd0b-4828-9d5c-006f66c1622e)

Further interesting stuff, there are no interesting things in the log to indicate what might be "waiting". In fact, from the logs provided by the container (I cannot find active logs _within_ the container), the `cancelExecution` endpoint is never even hit. I have attached the logs here 
[rundeck-rundeck-1_logs.txt](https://github.com/user-attachments/files/16252677/rundeck-rundeck-1_logs.txt)

---

My assumption is that I have misconfigured _something_ somewhere. I have no idea what as I really haven't done much yet. But the fact that a failed job doesn't stop and the endpoint to kill it doesn't register, that tells me I am likely doing something wrong. 

Any help would be greatly appreciated :)

**Comment by miversen33** [comment]
Some other interesting additional information I have dug up.

This issue came up when I attempted to migrate the rundeck stack (the rundeck container and the postgres database) to a new server.

I have tested the exact same job (failing in the exact same way) on my old instance of rundeck and it fails as expected. It doesn't get stuck in this weird state.

I copied the exact image over to the new server [version 5.1.0](https://hub.docker.com/layers/rundeck/rundeck/5.0.1/images/sha256-f3f65561845811596d043f7fb7cb8f8d1c010e32f208a8e1e6b65cf792106fa7?context=explore) and am experiencing the same behavior as before.

At this point, I have tested the failing job in the old server, exported the entire container to the new server and it fails with this bizarre "stuck" state. I'm wondering if there is some shenanigans going on below the hood but I am not sure how to get better logging to see what that might be.
